lgc/include/lgc/state/PalMetadata.h

```diff
@@ -171,6 +171,14 @@ class PalMetadata {
   // Set userDataLimit to maximum
   void setUserDataLimit();
 
+  // Returns true of the some of the user data nodes are spilled.
+  bool userDataNodesAreSpilled() const { return m_spillThreshold->getUInt() != USER_DATA_SPILL_IS_UNUSED; }
+
+  // Returns true of the setting in the PAL meta data require all of the user data nodes.
+  bool pipelineRequiresAllUserData() const;
+
+  const uint64_t USER_DATA_SPILL_IS_UNUSED = UINT_MAX;
```

Comment:
I find this line a bit confusing. Is the intention here to have the bit pattern of all ones (IE the max value for uint64_t) or the max value for uint32_t? `UINT_MAX` seems like the latter.

Reply:
The intention is to have the maximum possible value for this entry, which is determined by how PAL reads the value.  LLPC has always treated is as a 32-bit value, even if PAL only reads 16 bits (https://github.com/GPUOpen-Drivers/llpc/pull/1219#discussion_r615483442).  I've kept that the same.

The type is `uint64_t` because that is the type in the PAL metadata. 

I've renamed the variable to better represent the value it holds.

Reply:
Thanks, the comment makes it much clearer.
One more nit: since this is a compile-time constant, consider making it `static constexpr USER_....`. This way it would be clear that constructors cannot assign a different value and it won't be taking up space as a field of this class.

Reply:
Done.